External/Plugins/ProjectManager/Actions/FileActions.cs

```diff
@@ -385,79 +385,17 @@ public void Delete(string[] paths)
-                            List<string> filesToDelete = new List<string>(files.Count);
-                            foreach (KeyValuePair<string, int> filePair in files)
-                            {
-                                bool isSafeToDelete = true;
-
-                                foreach (KeyValuePair<string, int> dirPair in dirs)
-                                {
-                                    if (filePair.Key.Contains(dirPair.Key))
-                                    {
-                                        isSafeToDelete = false;
-                                        break;
-                                    }
-                                }
-
-                                if (isSafeToDelete)
-                                {
-                                    filesToDelete.Add(filePair.Key);
-                                }
-                            }
-
-                            // Clear out the directories first.
-                            foreach (KeyValuePair<string, int> dirPair in dirs)
-                            {
-                                if (!FileHelper.Recycle(dirPair.Key))
-                                {
-                                    String error = TextHelper.GetString("FlashDevelop.Info.CouldNotBeRecycled");
-                                    throw new Exception(error + " " + dirPair.Key);
-                                }
-                                OnFileDeleted(dirPair.Key);
-                            }
-
-                            // Then do the remaining files.
-                            foreach (string file in filesToDelete)
-                            {
-                                if (!FileHelper.Recycle(file))
-                                {
-                                    String error = TextHelper.GetString("FlashDevelop.Info.CouldNotBeRecycled");
-                                    throw new Exception(error + " " + file);
-                                }
-                                OnFileDeleted(file);
-                            }
-                        }
-                        else
-                        {
-                            // There are no directories being deleted so it's safe to just kill everything.
-                            foreach (string path in paths)
+                            if (File.Exists(path) || Directory.Exists(path))
                             {
                                 if (!FileHelper.Recycle(path))
                                 {
                                     String error = TextHelper.GetString("FlashDevelop.Info.CouldNotBeRecycled");
                                     throw new Exception(error + " " + path);
                                 }
-                                OnFileDeleted(path);
                             }
+                            OnFileDeleted(path);
```

Comment:
I'm a little unsure what how to handle OnFileDeleted(). Should it be inside the exist checks or outside? Technically if the file no longer exists then it _was_ deleted by the directory delete operation. So should it still call OnFileDeleted() on those files?